### PR TITLE
Laravel 10.x Compatibility

### PR DESCRIPTION
**Pull request by laravel-shift**
This is an automated pull request from [Shift](https://laravelshift.com) to update your package code and dependencies to be compatible with Laravel 9.x.

**Before merging**, you need to:

- Checkout the `l10-compatibility` branch
- Review **all** comments for additional changes
- Thoroughly test your package

If you do find an issue, please report it by commenting on this PR to help improve future automation.

### CODE REVIEW

**Comment by laravel-shift** [comment]
:warning: Shift [updated your build matrix](https://github.com/makeabledk/laravel-eloquent-status/pull/21/commits/b230ad807e16f79477456f7661330b683698b4f6?w=1) for Laravel 10. However, since you are supporting older versions of Laravel, you need to add the following exclusions:

```yaml
exclude:
  - laravel: 10.*
    php: 8.0
```

**Comment by laravel-shift** [comment]
:warning: Shift detected GitHub Actions which run jobs using a version `matrix`. Shift attempted to [update your configuration](https://github.com/makeabledk/laravel-eloquent-status/pull/21/commits/b230ad807e16f79477456f7661330b683698b4f6?w=1) for Laravel 10. However, you should review these changes to ensure the desired combination of versions are built for your package.

**Comment by laravel-shift** [comment]
:alembic: **Using this package?** If you would like to help test these changes or believe them to be compatible, you may update your project to reference this branch.

To do so, temporarily add Shift's fork to the `repositories` property of your `composer.json`:

```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/laravel-shift/laravel-eloquent-status.git"
        }
    ]
}
```

Then update your dependency constraint to reference this branch:

```json
{
    "require": {
        "makeabledk/laravel-eloquent-status": "dev-l10-compatibility",
    }
}
```

Finally, run: `composer update`

**Comment by mbardelmeijer** [comment]
@rasmuscnielsen would you mind taking a look at this PR and merging/tagging if acceptable?

Have submitted a PR to `makeabledk/laravel-query-kit` with Laravel 10 support: https://github.com/makeabledk/laravel-query-kit/pull/24

File: composer.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     },
     "require": {
         "php": "^8.0",
-        "illuminate/support": "^8.80|^9.0",
+        "illuminate/support": "^8.80|^9.0|^10.0",
         "makeabledk/laravel-querykit": "^3.0"
```
Comment (rasmuscnielsen) [inline composer.json]:
```suggestion
        "makeabledk/laravel-querykit": "^3.0|^4.0"
```

File: composer.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     },
     "require": {
         "php": "^8.0",
```
Comment (rasmuscnielsen) [inline composer.json]:
```suggestion
        "php": "^8.1",
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ jobs:
             fail-fast: false
             matrix:
                 php: [8.0, 8.1]
```
Comment (rasmuscnielsen) [inline .github/workflows/test.yml]:
```suggestion
                php: [8.1, 8.2]
```

File: composer.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
...
-        "illuminate/support": "^8.80|^9.0",
+        "illuminate/support": "^8.80|^9.0|^10.0",
```
Comment (rasmuscnielsen) [inline composer.json]:
```suggestion
        "illuminate/support": "^9.0|^10.0",
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -13,8 +13,8 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                php: [8.0, 8.1]
-                laravel: [8.*, 9.*]
+                php: [8.1, 8.2]
+                laravel: [8.*, 9.*, 10.*]
```
Comment (rasmuscnielsen) [inline .github/workflows/test.yml]:
```suggestion
                laravel: [9.*, 10.*]
```

File: composer.json [hunk 6]
```diff
@@ -7,9 +7,9 @@
...
     "require": {
-        "php": "^8.0",
-        "illuminate/support": "^8.80|^9.0",
-        "makeabledk/laravel-querykit": "^3.0"
+        "php": "^8.1",
+        "illuminate/support": "^9.0|^10.0",
+        "makeabledk/laravel-querykit": "^3.0|^4.0"
     },
     "require-dev": {
         "laravel/laravel": "^8.0|^9.0",
```
Comment (rasmuscnielsen) [inline composer.json]:
```suggestion
        "laravel/laravel": "^9.0|^10.0",
```

**Comment by rasmuscnielsen** [comment]
Cheers for the reminder @mbardelmeijer - merging and tagging

**Pull request merged**